src/main/scala/codacy/plugins/test/runner/ToolRunner.scala

```diff
@@ -0,0 +1,111 @@
+package codacy.plugins.test.runner
+
+import java.nio.file.{Path, Paths}
+
+import better.files._
+import codacy.plugins.test.DockerHelpers
+import com.codacy.analysis.core.model._
+import com.codacy.plugins.api
+import com.codacy.plugins.api.PatternFormatImplicits._
+import com.codacy.plugins.api.results.Tool
+import play.api.libs.json.{JsArray, JsError, JsResult, JsSuccess, JsValue, Json, Reads}
+
+import sys.process._
+
+object ToolRunner {
+  implicit private val resultReads = new Reads[ToolResult] {
+
+    def reads(json: JsValue): JsResult[ToolResult] = {
+      val patternIdOpt = (json \ "patternId").asOpt[String]
+      val lineOpt = (json \ "line").asOpt[Int]
+      val message = (json \ "message").as[String]
+      val filename = (json \ "filename").as[String]
+
+      (patternIdOpt, lineOpt) match {
+        case (None, None) => JsSuccess(FileError(Paths.get(filename), message))
+        case (Some(patternId), Some(line)) =>
+          JsSuccess(
+            Issue(api.results.Pattern.Id(patternId),
+                  Paths.get(filename),
+                  message = Issue.Message(message),
+                  api.results.Result.Level.Info,
+                  None,
+                  LineLocation(line))
+          )
+        case _ => JsError(s"Wrong result format: ${Json.prettyPrint(json)}")
+      }
+    }
+  }
+
+  def run(dockerImage: String, srcDir: File, files: Set[Path], configuration: Configuration): Seq[ToolResult] =
+    DockerHelpers.usingDocsDirectoryInDockerImage(dockerImage) { docsDirectory =>
+      (for {
+        codacyrcFile <- File.temporaryFile(parent = Some(File.root / "tmp"))
+        patternsJsonString = (docsDirectory.toScala / "patterns.json").contentAsString
+        patternsJson = Json.parse(patternsJsonString)
+        toolName = (patternsJson \ "name").get.as[String]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of get on optional type.](https://app.codacy.com/gh/codacy/codacy-plugins-test/pullRequest?prid=5569494)